data/notebooks/ingredients.py

```diff
@@ -0,0 +1,689 @@
+    style=style,
+)
+w_organic_default_search = ipywidgets.Text(
+    placeholder="flour conventio*", layout=layoutL, style=style
+)
+w_organic_simple_ingredient_default = ipywidgets.Select(
+    placeholder="Process",
+    layout=layoutW,
+    options=[""],
+    rows=1,
+    style=style,
+)
+w_organic_variant_search = ipywidgets.Text(
+    placeholder="flour organic", layout=layoutL, style=style
+)
+w_organic_simple_ingredient_variant = ipywidgets.Select(
+    placeholder="Process",
+    tooltip="Click to add or update the ingredient",
+    style=style,
+    options=[""],
+    rows=1,
+    layout=layoutW,
+)
+# default coef for the beyond-lca indicators
+w_organic_agrodiv = ipywidgets.FloatSlider(
+    placeholder="Agro-diversity coefficient",
+    style=style,
+    min=0,
+    max=1,
+    step=0.05,
+    layout=layoutW,
+)
+w_organic_agroeco = ipywidgets.FloatSlider(
+    placeholder="Agro-ecology coefficient",
+    min=0,
+    max=1,
+    step=0.05,
+    style=style,
+    layout=layoutW,
+)
+w_organic_animal_welfare = ipywidgets.FloatSlider(
+    placeholder="Animal Welfare coefficient",
+    min=0,
+    max=1,
+    step=0.05,
+    style=style,
+    layout=layoutW,
+)
+## code for BleuBlanCoeur process if any
+w_bleu_blanc_coeur_search = ipywidgets.Text(
+    placeholder="bleu blanc", layout=layoutL, style=style
+)
+w_bleu_blanc_coeur = ipywidgets.Select(
+    placeholder="Process",
+    style=style,
+    options=[""],
+    rows=1,
+    layout=layoutW,
+)
+
+
+# buttons
+savebutton = ipywidgets.Button(
+    description="Save",
+    button_style="warning",  # 'success', 'info', 'warning', 'danger' or ''
+    tooltip="Add or update the ingredient",
+    icon="check",
+)
+delbutton = ipywidgets.Button(
+    description="Delete",
+    button_style="danger",  # 'success', 'info', 'warning', 'danger' or ''
+    tooltip="Delete the ingredient with the 'id' field above",
+    icon="trash",
+)
+getbutton = ipywidgets.Button(
+    description="Get",
+    button_style="success",  # 'success', 'info', 'warning', 'danger' or ''
+    tooltip="Fill the form with ingedient from the 'id' field",
+    icon="down-to-bracket",
+)
+resetbutton = ipywidgets.Button(
+    description="Reset from branch",
+    button_style="success",  # 'success', 'info', 'warning', 'danger' or ''
+    tooltip="Reset the ingredients to the branch state",
+    icon="sparkles",
+)
+commitbutton = ipywidgets.Button(
+    description="Publish",
+    button_style="danger",  # 'success', 'info', 'warning', 'danger' or ''
+    tooltip="Commit the ingredients into the branch",
+    icon="code-commit",
+)
+out = ipywidgets.Output()
+
+
+def list_ingredients():
+    ingredients = read_ingredients()
+    with open(INGREDIENTS_TEMP) as fp:
+        display(
+            Markdown(f"# List of {len(ingredients)} ingredients:"),
+            pandas.DataFrame(ingredients.values(), columns=list(FIELDS.values())),
+            Markdown(f"# Resulting JSON file:"),
+            json.load(fp),
```

Comment:
This renders a Python dict:

![image](https://user-images.githubusercontent.com/41547/233354122-2137eb93-6d2c-4dcc-b1f4-d0dc991d502b.png)

Ideally we should render a JSON string so we could easily test it eg. with the Web client.

Reply:
done